plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PluginTaskScheduler, TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { GitLabIntegration, ScmIntegrations } from '@backstage/integration';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-backend';
+import * as uuid from 'uuid';
+import { Logger } from 'winston';
+import {
+  GitLabClient,
+  GitlabProviderConfig,
+  paginated,
+  readGitlabConfigs,
+} from '../lib';
+import { GitLabGroup, GitLabUser } from '../lib/types';
+import {
+  ANNOTATION_LOCATION,
+  ANNOTATION_ORIGIN_LOCATION,
+  Entity,
+  UserEntity,
+  GroupEntity,
+} from '@backstage/catalog-model';
+import { merge } from 'lodash';
+
+type Result = {
+  scanned: number;
+  matches: GitLabUser[];
+};
+
+type GroupResult = {
+  scanned: number;
+  matches: GitLabGroup[];
+};
+
+/**
+ * Discovers entity definition files in the groups of a Gitlab instance.
+ * @public
+ */
```

Comment:
Hmmm copy-paste. Maybe something like
```suggestion
/**
 * Discovers users and groups from a Gitlab instance.
 * @public
 */
```
would be better?